libs/src/services/creatorNode/index.js

```diff
@@ -174,22 +172,27 @@ class CreatorNode {
 
     const [trackContentResp, coverArtResp] = await Promise.all(uploadPromises)
     metadata.track_segments = trackContentResp.track_segments
+    const sourceFile = trackContentResp.source_file
```

Comment:
if sourceFile is not defined here, we can fail faster and avoid the unnecessary uploadtrackMetadata call. thoughts?

Reply:
I think I would prefer that the client be stupider here and just pass through what it gets back. I think it gets more dangerous the more we add checks that all the inputs and retvals from various server calls are "correct." it basically means that the client has to semantically understand what "source file is" when it really shouldn't need to imo

Reply:
i agree in principle, but in practice - the libs are the api to backend services and are designed together so don't see reason to not optimize when we know the outcome. similar to how the download route knows that creatornodes will return different response headers than other ipfs gateways. but i don't feel strongly about this, as it's a cheap call anyway

Reply:
aight, ill throw some error in this case. easy enough. fwiw I feel like it's a bit of an anti-pattern (tightly coupling the apis) https://en.wikipedia.org/wiki/Coupling_(computer_programming) 
e.g. if we change the interface of the track content resp to sometimes not return a source file (or return an empty one), but be valid, the change has to cascade to libs

Reply:
agreed on this being an anti pattern, but it's the pattern used in libs so until we can properly decouple may as well take advantage of it. but also if we can incrementally decouple that's also valuable. either way, nbd

Reply:
👍